src/ThemeJsonExtractor.php

```diff
@@ -0,0 +1,175 @@
+						continue;
+					}
+
+					// Whole path will be [ 'settings', 'blocks', 'core/paragraph', 'color', 'palette' ].
+					$whole_path  = array_merge( $base_path, array( $node_name ), $data_path );
+					$translation = $translations->insert( $context, $original );
+					$translation->addReference( $file );
+				}
+			} else {
+				$array_to_translate = self::_wp_array_get( $theme_json, $path, null );
+				if ( is_null( $array_to_translate ) ) {
+					continue;
+				}
+
+				$translation = $translations->insert( $context, $original );
+				$translation->addReference( $file );
+			}
+		}
+	}
+
+	private static function read_json_file( $file_path ) {
+		$config = array();
+		if ( $file_path ) {
+			$decoded_file = json_decode(
+				file_get_contents( $file_path ),
+				true
+			);
+
+			$json_decoding_error = json_last_error();
+			if ( JSON_ERROR_NONE !== $json_decoding_error ) {
+				WP_CLI::debug( "Error when decoding {$file_path}", 'make-pot' );
+
+				return $config;
+			}
+
+			if ( is_array( $decoded_file ) ) {
+				$config = $decoded_file;
+			}
+		}
+		return $config;
+	}
+
+	private static function get_fields_to_translate() {
+		if ( null === self::$theme_json_i18n ) {
+			$file_structure        = self::read_json_file( __DIR__ . '/theme-i18n.json' );
+			self::$theme_json_i18n = self::extract_paths_to_translate( $file_structure );
+		}
+		return self::$theme_json_i18n;
+	}
+
+	private static function extract_paths_to_translate( $i18n_partial, $current_path = array() ) {
+		$result = array();
+		foreach ( $i18n_partial as $property => $partial_child ) {
+			if ( is_numeric( $property ) ) {
+				foreach ( $partial_child as $key => $context ) {
+					return array(
+						array(
+							'path'    => $current_path,
+							'key'     => $key,
+							'context' => $context,
+						),
+					);
+				}
+			}
+			$result = array_merge(
+				$result,
+				self::extract_paths_to_translate( $partial_child, array_merge( $current_path, array( $property ) ) )
+			);
+		}
+		return $result;
+	}
+
+	/**
+	 * Accesses an array in depth based on a path of keys.
+	 *
+	 * It is the PHP equivalent of JavaScript's `lodash.get()` and mirroring it may help other components
+	 * retain some symmetry between client and server implementations.
+	 *
+	 * Example usage:
+	 *
+	 *     $array = array(
+	 *         'a' => array(
+	 *             'b' => array(
+	 *                 'c' => 1,
+	 *             ),
+	 *         ),
+	 *     );
+	 *     _wp_array_get( $array, array( 'a', 'b', 'c' ) );
+	 *
+	 * @param array $array   An array from which we want to retrieve some information.
+	 * @param array $path    An array of keys describing the path with which to retrieve information.
+	 * @param mixed $default The return value if the path does not exist within the array,
+	 *                       or if `$array` or `$path` are not arrays.
+	 *
+	 * @return mixed The value from the path specified.
+	 */
+	private function _wp_array_get( $array, $path, $default = null ) {
```

Comment:
This is an utility that I've ported from WordPress core, so wp-cli doesn't depend on it being present.